x-pack/plugins/uptime/public/components/functional/charts/checks_chart.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  AreaSeries,
+  Axis,
+  Chart,
+  getAxisId,
+  getSpecId,
+  Position,
+  Settings,
+  ScaleType,
+  timeFormatter,
+} from '@elastic/charts';
+import { EuiPanel, EuiTitle } from '@elastic/eui';
+import React, { useContext } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { StatusData } from '../../../../common/graphql/types';
+import { getChartDateLabel } from '../../../lib/helper';
+import { UptimeSettingsContext } from '../../../contexts';
+import { getColorsMap } from './get_colors_map';
+
+interface ChecksChartProps {
+  /**
+   * The color that will be used for the area series displaying "Down" checks.
+   */
+  dangerColor: string;
+  /**
+   * The timeseries data displayed in the chart.
+   */
+  status: StatusData[];
+  /**
+   * The color that will be used for the area series displaying "Up" checks.
+   */
+  successColor: string;
+}
+
+/**
+ * Renders a chart that displays the total count of up/down status checks over time
+ * as a stacked area chart.
+ * @param props The props values required by this component.
+ */
+export const ChecksChart = ({ dangerColor, status, successColor }: ChecksChartProps) => {
+  const upSeriesSpecId = getSpecId('Up');
+  const downSeriesSpecId = getSpecId('Down');
+  const { absoluteStartDate, absoluteEndDate } = useContext(UptimeSettingsContext);
+
+  const upString = i18n.translate('xpack.uptime.monitorCharts.checkStatus.series.upCountLabel', {
+    defaultMessage: 'Up count',
+  });
+  const downString = i18n.translate(
+    'xpack.uptime.monitorCharts.checkStatus.series.downCountLabel',
+    {
+      defaultMessage: 'Down count',
+    }
+  );
+
+  return (
+    <React.Fragment>
+      <EuiTitle size="xs">
+        <h4>
+          <FormattedMessage
+            id="xpack.uptime.monitorCharts.checkStatus.title"
+            defaultMessage="Check status"
+          />
+        </h4>
+      </EuiTitle>
+      <EuiPanel>
+        <Chart renderer="canvas">
+          {
+            // @ts-ignore
+            <Settings
+              domain={{ absoluteStartDate, absoluteEndDate }}
```

Comment:
It's a bit weird. there is no `domain` in settings, you can use the `xDomain` prop with `{min?:number, max?:number}`  The `ts-ignore` also can create some issue in the future

Reply:
Thanks for pointing this out, 100% correct.

Resolved by https://github.com/elastic/kibana/pull/34313/commits/ba697bd0952d4c336e60ded3950af8600f19a596